### PR TITLE
Vision ready check

### PR DESCRIPTION
**Pull request by OliverW10**
added a function to the vision component to check the status of vision for the leds

### CODE REVIEW

File: components/vision.py [hunk 1]
```diff
@@ -12,4 +14,13 @@ def get_vision_data(self) -> Tuple[float, float, float]:
         """Returns a tuple containing the distance (metres), angle (radians), and timestamp (time.monotonic)
         If it can't get info, it returns [None, None, None]
         """
-        return self.entry.getDoubleArray([None, None, None])
+        self.data = self.entry.getDoubleArray([None, None, None])
+        return self.data
+
+    def is_ready(self) -> int:
+        if abs(self.data[2] - time.monotonic()) > 0.5:
```
Comment (auscompgeek) [inline components/vision.py]: This... doesn't seem correct. How are you guaranteeing the clock on the RPi is the same as the clock on the RIO?

Comment (OliverW10) [reply components/vision.py]: i have now changed it

File: components/vision.py [hunk 2]
```diff
@@ -1,15 +1,33 @@
...
+    def is_ready(self) -> int:
+        if self.data[2] == self.lastTime:  # if it gets the same time twice
+            self.repeats += 1
+        else:
+            self.repeats = 0
+        self.lastTime = self.data[2]
+
+        if self.repeats >= 5:  # if it has got the same data for five loops
+            return 0  # no target
+        if abs(self.data[1]) < math.degrees(5):
```
Comment (auscompgeek) [inline components/vision.py]: This will break if there is no vision data.

Comment (OliverW10) [reply components/vision.py]: There is now a check for None

File: components/vision.py [hunk 3]
```diff
@@ -7,9 +7,33 @@ def __init__(self) -> None:
...
     def get_vision_data(self) -> Tuple[float, float, float]:
         """Returns a tuple containing the distance (metres), angle (radians), and timestamp (time.monotonic)
-        If it can't get info, it returns [None, None, None]
+        If it can't get the entry, it returns [None, None, None]
+
+        If the pi vision cant see the target it will not send to network tables, meaning this will continue
+        to return the most recently seen target position and the time it was seen
         """
-        return self.entry.getDoubleArray([None, None, None])
+        if self.entry.exists():
```
Comment (auscompgeek) [inline components/vision.py]: ... isn't this saying exactly the same thing twice?

Comment (OliverW10) [reply components/vision.py]: yes, i didn't realise they were doing the same thing and have removed the entry.exists check

**Pull request merged**